File: ios/Poof/BusinessWorkspace.swift
import SwiftUI

// Business tier: on-device workspace store.
// Name, logo char, accent color, per-peer role. All persisted in UserDefaults.

enum BusinessWorkspace {
    static let keyName = "poof.workspace.name"
    static let keyLogo = "poof.workspace.logo"
    static let keyColor = "poof.workspace.color"
    static let keyRoles = "poof.workspace.roles"

    static let colorPresets: [String] = [
        "#5B8BFF", "#8B7EFF", "#3FBF7F", "#FF7A9C", "#FFBF3C", "#FF5C52"
    ]

    static let defaultColor = "#5B8BFF"
    static let defaultLogo = "P"

    static let roles: [String] = ["Admin", "Member", "Guest"]

    static var name: String {
        get { UserDefaults.standard.string(forKey: keyName) ?? "" }
        set {
            let t = newValue.trimmingCharacters(in: .whitespacesAndNewlines)
            if t.isEmpty {
                UserDefaults.standard.removeObject(forKey: keyName)
            } else {
                UserDefaults.standard.set(t, forKey: keyName)
            }
        }
    }

    static var logo: String {
        get { UserDefaults.standard.string(forKey: keyLogo) ?? defaultLogo }
        set {
            let t = String(newValue.prefix(2)).trimmingCharacters(in: .whitespacesAndNewlines)
            let v = t.isEmpty ? defaultLogo : t
            UserDefaults.standard.set(v, forKey: keyLogo)
        }
    }

    static var color: String {
        get { UserDefaults.standard.string(forKey: keyColor) ?? defaultColor }
        set { UserDefaults.standard.set(newValue, forKey: keyColor) }
    }

    static var accent: Color {
        Color(hex: color)
    }

    static func role(for peerId: String) -> String {
        rolesDict[peerId] ?? "Member"
    }

    static func setRole(_ role: String, for peerId: String) {
        var dict = rolesDict
        dict[peerId] = role
        setRolesDict(dict)
    }

    static var rolesDict: [String: String] {
        guard let data = UserDefaults.standard.data(forKey: keyRoles),
              let dict = try? JSONDecoder().decode([String: String].self, from: data)
        else { return [:] }
        return dict
    }

    private static func setRolesDict(_ dict: [String: String]) {
        if let data = try? JSONEncoder().encode(dict) {
            UserDefaults.standard.set(data, forKey: keyRoles)
        }
    }
}

// init(hex:) is now defined in PoofColors.swift (single source of truth).

File: docs/pc/src/business-workspace.js
// Business tier: on-device workspace store (mirror of ios BusinessWorkspace).
// Name, logo, accent color, per-peer role. All persisted in localStorage.

const KEY_NAME  = 'poof.workspace.name';
const KEY_LOGO  = 'poof.workspace.logo';
const KEY_COLOR = 'poof.workspace.color';
const KEY_ROLES = 'poof.workspace.roles';

export const COLOR_PRESETS = ['#5B8BFF', '#8B7EFF', '#3FBF7F', '#FF7A9C', '#FFBF3C', '#FF5C52'];
export const DEFAULT_COLOR = '#5B8BFF';
export const DEFAULT_LOGO  = 'P';
export const ROLES = ['Admin', 'Member', 'Guest'];

export function getName()  { return localStorage.getItem(KEY_NAME) || ''; }
export function getLogo()  { return localStorage.getItem(KEY_LOGO) || DEFAULT_LOGO; }
export function getColor() { return localStorage.getItem(KEY_COLOR) || DEFAULT_COLOR; }

export function setName(v) {
  const t = String(v || '').trim();
  if (t) localStorage.setItem(KEY_NAME, t);
  else localStorage.removeItem(KEY_NAME);
}
export function setLogo(v) {
  const t = String(v || '').trim().slice(0, 2);
  localStorage.setItem(KEY_LOGO, t || DEFAULT_LOGO);
}
export function setColor(v) {
  localStorage.setItem(KEY_COLOR, v || DEFAULT_COLOR);
}

export function getRoles() {
  try { return JSON.parse(localStorage.getItem(KEY_ROLES) || '{}'); }
  catch { return {}; }
}
export function getRole(peerId) { return getRoles()[peerId] || 'Member'; }
export function setRole(peerId, role) {
  const r = getRoles();
  r[peerId] = role;
  localStorage.setItem(KEY_ROLES, JSON.stringify(r));
}

export function workspaceState() {
  return { name: getName(), logo: getLogo(), color: getColor(), roles: getRoles() };
}
